src/imu_3DM_GX3_25.cpp

```diff
@@ -75,12 +75,26 @@ bool Imu3DM_GX3_25::open_usb_port(int bauderate)
 
 bool Imu3DM_GX3_25::receive_message(ImuMsg& msg, bool stream_mode)
 {
-  if(!usb_stream_.read_device(msg.reply_, stream_mode))
+  ssize_t read_bytes = usb_stream_.read_maybe_device(msg.reply_, stream_mode);
+  if(read_bytes < 0)
   {
     rt_printf("Imu3DM_GX3_25::receive_message(): [Error] "
               "message badly received\n");
   }
 
+  // rt_printf("\nBREAK debug received message of size %ld, expected %ld\n\n", asdf, msg.reply_.size());
+  if(read_bytes != static_cast<ssize_t>(msg.reply_.size()))
```

Comment:
would it make sense to change this into a `while` loop? `while (total_bytes < expected_bytes):`